TagsterWebAPI/src/Tagster.Infrastructure/Services/AppInitializer.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace Tagster.Infrastructure.Services
+{
+    internal class AppInitializer : IHostedService
+    {
+        private readonly IServiceProvider _serviceProvider;
+
+        public AppInitializer(IServiceProvider serviceProvider)
+            => _serviceProvider = serviceProvider;
+
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            var dbContextTypes = AppDomain.CurrentDomain.GetAssemblies()
+               .SelectMany(x => x.GetTypes())
+               .Where(x => typeof(DbContext).IsAssignableFrom(x) && !x.IsInterface && x != typeof(DbContext));
+
+            using var scope = _serviceProvider.CreateScope();
+            foreach (var dbContextType in dbContextTypes)
+            {
+                if (scope.ServiceProvider.GetService(dbContextType) is not DbContext dbContext)
```

Comment:
Doesn't if expression always be false?
I think that it's impossible to pass this condition because above you have `Where `which ensures that
`x => typeof(DbContext).IsAssignableFrom(x)`

Unless I missed sth.

Reply:
ok, I've got it :) there is a possibility to have class which is not registered in DI container, so skip this